group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2406,6 +2411,17 @@ private boolean hasMemberSubscriptionChanged(
             if (!updatedMember.subscribedTopicRegex().equals(member.subscribedTopicRegex())) {
                 log.debug("[GroupId {}] Member {} updated its subscribed regex to: {}.",
                     groupId, memberId, updatedMember.subscribedTopicRegex());
+                // If the regular expression has changed, we compile it to ensure that
+                // its syntax is valid.
+                if (updatedMember.subscribedTopicRegex() != null) {
+                    try {
+                        Pattern.compile(updatedMember.subscribedTopicRegex());
+                    } catch (PatternSyntaxException ex) {
+                        throw new InvalidRegularExpression(
+                            String.format("SubscribedTopicRegex `%s` is not a valid regular expression: %s.",
+                                updatedMember.subscribedTopicRegex(), ex.getDescription()));
+                    }
+                }
```

Comment:
would it be better maybe to encapsulate this in a kind of `throwIfInvalidRegularExpression`? (it aligns with the validation pattern we have in the mgr, and it would probably be the place to extend with caching changes)